modules/nlu/src/backend/engine.ts

```diff
@@ -113,11 +122,31 @@ export default class Engine implements NLUEngine {
       ctxToTrain
     }
 
-    // Model should be build here, Trainer should not have any idea of how this is stored
-    // Error handling should be done here
-    let model = await Trainer(input, Engine.tools)
+    let model: Partial<UntrainedModel> = {
+      startedAt: new Date(),
+      languageCode: input.languageCode,
+      data: {
+        input
+      }
+    }
+
+    try {
```

Comment:
train function is already super bloated .. and I feel like we're adding even more complexity to it . We might want to do another refactor PR dedicated to this immense train function. Keep this in mind for standalone NLU project :)

Reply:
Yes you're right, I will open a new PR later only for that